### ISSUE TITLE
"Timed out waiting for the flyout to open and populate."

### ISSUE THREAD

**Issue by caseych4n** [Hari-Nagarajan/fairgame#717]
**Describe the bug**
Not sure if this is a bug or not, but this only happens for pages with the "see all buying options" window in the screenshot
![Screenshot 2021-04-22 012026](https://user-images.githubusercontent.com/58901241/115681008-f62ad480-a308-11eb-83be-55ebd2d5348f.png)

**To Reproduce**
Run the code with these ASIN's as examples:
This issue happens with multiple items here are some example asins
B08VLLBFR3 (zotac rtx 3080 white)
B08P7ZKQPP (zotac rtx 3070 white)

this is what the command prompt shows when the error shows up:
![timedoutzotac](https://user-images.githubusercontent.com/58901241/115681318-443fd800-a309-11eb-9958-6c0eafcb643c.png)

After I clicked, "see all buying options," the code would run through and then refresh then run into the same problem again.

**Desktop (please complete the following information):**
 - OS: Windows 10 64bit Pro
 - Browser: Google Chrome (latest)

the bot runs fine on the items that are completely out of stock, but not on one's that have scalpers and another third party amazon sellers.

**Comment by HaloGeekster** [comment]
@casetch4n I have the same issue occasionally. (Though I assume its my fault for running 12+ instances at the same time.) Watching the browser window, I see the fly-out pop out for a fraction of a second and then immediately close itself while the dark overlay on the rest of the page remains I fixed it once by clearing browser-data, and another time by 'End Task'ing some extra instances of pipenv.exe that remained open in background  @processes for some reason.

**Comment by tgadberry1** [comment]
I noticed this issue for the first time today when I was setting up a 3rd machine to watch additional items. Namely, the 5900X but not 3060Ti's gave me this issue. As @HaloGeekster mentioned, it may be a botting issue. When it happens on fair game,  it happens for me just the same when I manually try to check all buying options. However, open that same page in an incognito tab and it works just fine.

**Comment by IvanShift** [comment]
> Edit: It could also be an anti-botting measure, I've noticed I cant access my account page from browser anymore. I just get a "Sorry, we couldn't find that page" with a dog.

I called Amazon support. They have no idea why this is happening or what to do. But they gave me this  link: https://amazon.com/access 
It shows the Amazon site as a mobile version where you can access to "Account"-page.

**Comment by jobes911** [comment]
I got the same problem,   Captcha  and    the See all buyin option button  / slide on the right side of the screen remain empty and  the bot  doesnt work.

**Comment by tgadberry1** [comment]
Curious. How many of you were running multiple instances on your account (either on one or multiple devices), and how low did you set the delay?

**Comment by tgadberry1** [comment]
> > Edit: It could also be an anti-botting measure, I've noticed I cant access my account page from browser anymore. I just get a "Sorry, we couldn't find that page" with a dog.
> 
> I called Amazon support. They have no idea why this is happening or what to do. But they gave me this link: https://amazon.com/access

To be fair, I doubt your standard Amazon auto er service rep is versed enough to know about all of the anti-bottling measures the devs slipped in. I’m going to kill all instances today and give it 24hrs to see if it resets.

**Comment by tgadberry1** [comment]
> To be fair, I doubt the standard Amazon customer service rep is versed enough to know about all anti-bottling measures the devs may have slipped in. I’m going to kill all instances today and give it 24hrs to see if it resets.

Update: Once all instances were stopped, my amazon looks like it has returned to normal within an hour or two. For now, the solution seems to be to exercise caution and don't get too greedy. Amazon may have implemented these soft blocks to throttle bot traffic/efficacy.  Consider 1) increasing the delay between ASIN cycles, 2) don't run multiple devices from multiple locations, and 3) keep it to one or two instances of the program. If you happen to find more precise tolerances please share them!

**Comment by JCrowe86** [comment]
Having the same issue.   Only running one instance with 2 ASINs, 1.5sec delay.  Kept getting the error a lot yesterday.  Ran almost all day with no issue it but something triggered it a bit ago and now I can't get it to go away.

**Comment by tgadberry1** [comment]
Well, I said everything was back to normal a little too early. My account still seems to have a soft-ban. Looking back through older issues posted, this has been a very common issue. I can’t get a definite answer as to how long the limits will last. I would strongly recommend that the ReadMe include a warning early on about too much/too fast and how it can impact accounts. There are reports of it happening even with the default 3 second delay. I will update once my account is back to normal or if I have to go through Amazon support.

**Comment by DakkJaniels** [comment]
You are welcome to make a PR to write whatever warning you think is appropriate. We already tell people not to run multiple instances, but people don't seem to care.
![image](https://user-images.githubusercontent.com/6080734/115938775-c0513180-a469-11eb-9b71-51be3749b6c6.png)

**Comment by tgadberry1** [comment]
Ah, I interpreted “concurrently” as two instances on the same device. Semantic issue on my part.

Thanks! I’ll try to clarify when I put something together!

**Comment by Eternity-Soap** [comment]
I agree that this appears to be an anti-botting measure that's triggered by running multiple instances. I also got greedy and disregarded the warnings about doing so and am having this problem as well. Interestingly it only seems to affect my chrome browser; I have no difficulty looking at purchase options when using Edge manually. It also only seems to affect the offending account and leaves others coming from the same network alone.

**Comment by CptnFluffy** [comment]
I started having this problem yesterday. I was running two instances with a one second delay. I turned it off overnight and started it again this morning, but only one instance with a two second delay. So far it's been running for about five hours and there haven't been any problems. My moneys on anti-botting measure as well.

**Comment by CJoseFlores** [comment]
> I agree that this appears to be an anti-botting measure that's triggered by running multiple instances. I also got greedy and disregarded the warnings about doing so and am having this problem as well. Interestingly it only seems to affect my chrome browser; I have no difficulty looking at purchase options when using Edge manually. It also only seems to affect the offending account and leaves others coming from the same network alone.

It seems that amazon does not do any soft-banning by IP, and it's only per account. One thing I did note from others (including from a previous comment in this thread) is that you can check if you're soft banned by attempting to access the 'Account' page. Even if the fly-out works on some browsers, if you can't access the 'Account' page, you're still soft banned.

**Comment by tgadberry1** [comment]
In my experience, shadow bans clear ~24 hours after you halt all bots on
your account. There's a new "Disclaimer" section on the README that helps
bring the issue forward to new users.

**Comment by B3SAVAGE** [comment]
Just a quick update if people are still having issues, **clearing cookies** in the browser window that fairgame is running in should fix the issue mentioned above. At least it has been a solution that has worked in my case

**Comment by mlgvain** [comment]
> Just a quick update if people are still having issues, **clearing cookies** in the browser window that fairgame is running in should fix the issue mentioned above. At least it has been a solution that has worked in my case

I'm having this issue with 0.6.8
It used to not have a problem, then something happened over-night and it's giving me the same thing. 

I tried clearing the cookies, but that didn't seem to help.

**Comment by Scope666** [comment]
Also having this problem, started yesterday morning, I think Amazon is getting more aggressive with their anti-bot settings.  Clearing cookies does NOT help.  (soft-ban, you see the dogs when you click on the ACCOUNT page)

**Comment by mlgvain** [comment]
> Also having this problem, started yesterday morning, I think Amazon is getting more aggressive with their anti-bot settings. Clearing cookies does NOT help. (soft-ban, you see the dogs when you click on the ACCOUNT page)

Yeah, that's what I was getting, but oddly enough, if I used a different browser it let me go to my Account page with no issues.

**Comment by Scope666** [comment]
I'm seeing the same, they're probably soft-banning a combo of the IP address AND the user agent of the browser.

**Comment by BorisG491** [comment]
I had the same problem. I am getting around it by using VPN (different Amazon account, different IP).

**Comment by BorisG491** [comment]
It's not just IP an user agent. In my case I think it's the account as well. I can't get the "see all buying options" to work even on my phone's amazon app. Used a different IP, still not working. Using edge, still not working. Registered a new account and running from a different IP, it works again.

**Comment by rockstarblech** [comment]
So what's the fix for this ... You guys just keep making new accounts? Mine did this after less than 12hrs ......

**Comment by Scope666** [comment]
Using the --delay parameter fixed it for me, around 7 - 8 seconds seems to work.

**Comment by DeathsSatellite** [comment]
Had the same issue. An ipconfig/release and then ipconfig/renew seemed to work for me.

**Comment by Jay311xx** [comment]
Just happened to my only running one instance.  Cookies and ipconfig/release ipconfig/renew did not work.  First time using app for less than an hour.  I was only doing one item.  So maybe because it was spamming that item so fast I got a ban.  Any fix besides just waiting?  Guess I will just put some more items in next time I run it.

**Comment by Jay311xx** [comment]
Temp ban again today after running it for an hour.  Was doing 4 items this time so it wasn't because of spamming same item.  Think app is broken atm so once fixed I will start with 8 second delay and if that works keep going down 1 second.  Will post results.
